app/main/download_data.py

```diff
@@ -238,3 +241,82 @@ def process_api_response(query_params: dict) -> tuple:
         return abort(500), f"Unknown content type: {content_type}"
 
     return content_type, file_content
+
+
+def process_async_download(query_params: dict):
+    """process async download request to start the background process.
+    :param query_params: (dict): Query parameters for the API request.
+    """
+    request_url = (
+        Config.DATA_STORE_API_HOST
+        + "/trigger_async_download"
+        + ("?" + urlencode(query_params, doseq=True) if query_params else "")
+    )
+    requests.post(request_url, timeout=20)
+
+
+def get_presigned_url(filename: str):
+    """Get the presigned link for the short time to retrieve the file from s3 bucket.
+    :param filename (str): object name which needs to be retrieved from s3 if exists
+    Raises:ValueError: If object doest not exists in S3, it will raise an error.
+    Returns:Returns the response the API.
+    """
+    if not filename:
+        raise ValueError("filename is required")
+
+    response = get_response(Config.DATA_STORE_API_HOST, f"/get-presigned-url/{filename}")
+    return response.json()["presigned_url"]
+
+
+FileMetadata = namedtuple("FileMetadata", ["response_status_code", "formated_date", "file_format", "file_size_str"])
+
+
+def get_find_download_file_metadata(filename: str) -> FileMetadata:
+    """To get the object metadata from S3 using the ovject Key
+    :param filename (str): object name to get the metadata
+
+    Raises:
+        ValueError: If object doest not exists in S3, it will raise an error.
+
+    Returns: FileMetadata:
+             - Returns the last modified date,
+             -file format, and human-readable file size.
+    """
+    response = get_response(Config.DATA_STORE_API_HOST, f"/get-find-download-metadata/{filename}")
+    response_status_code = response.status_code
+
+    if response_status_code == 200:
```

Comment:
```suggestion
    if response.status_code == 200:
```

This extra variable doesn't do anything for us - it doesn't make the code more readable and it doesn't make anything faster. We can remove it.